### DISCUSSION TITLE
Curious about the aggregation algorithm

### DISCUSSION THREAD

**Discussion by cyclaero** [searx/searx#2614] [General]
Since 3 weeks, I am almost solely searching the internet via a private searx instance running on our gateway to the internet. I am really satisfied, and please don’t take this as a complaint, everything is perfect. Just out of curiosity, I would like to learn more about the aggregation algorithm.

Basically my questions are:
1. What are the criteria of result ordering?
2. Besides sponsored results, are there other kinds of results which become suppressed?
3. On subsumed same results from different search engines, the engines are named by tags, which doesn’t seem to be in alphabetical order, so which order is it then?
4. In most of my queries, the results from Duckduckgo and Qwant are prevalent over Google, which doesn’t match the expectation. Sometimes I see only one result from Google in the middle of a page full of results from the others. Again, I am not complaining, I am only curious about why this may happen.

**Comment by unixfox** [comment]
If I'm not mistaken your questions got answered in https://github.com/searx/searx/issues/1790

**Reply by cyclaero** [reply]
Yes, thank you very much, the first question and most important one is answered. So we may change the order by adding weight parameters to the engine’s entries in settings.yml. Currently, I got only two entries with a weight higher than one, namely `weight : 2`  for ddd and for wd, which explains to a certain extend the prevalence of Duckduckgo - I am happy with this and I will stay with this setting.
